include/library.h

```diff
@@ -263,6 +263,18 @@ class Library: public std::enable_shared_from_this<Library>
    */
   bool removeBookmark(const std::string& zimId, const std::string& url);
 
+  /**
+   * Resolve bookmarks
+   *
+   * @param source the source bookId of the bookmarks to resolve.
+                   If not present, we resolve all bookmarks pointing to an unknown book.
+   * @param destination the destination bookId to resolve te bookmark to.
+                        If not present, we will try to determine it using heuristics based on
+                        book name, flavour and date.
+   * @return The number of bookmarks created.
+   */
+  int resolveBookmarks(const std::string& source = "", std::string destination = "");
```

Comment:
Isn't "migrate" a better verb for this operation?

Reply:
Definitly

Reply:
Yes, I follow the @kelson42 proposition at first and made the same thought to myself during development, forget about it.